v2/tools/generator/internal/conversions/README.md

```diff
@@ -0,0 +1,568 @@
+# Property Conversions
+
+One of our challenges is generating the code to copy losslessly between different versions of each resource.
+
+## How it works
+
+The primary entry point is [`CreateTypeConversion`](https://github.com/Azure/azure-service-operator/blob/main/v2/tools/generator/internal/conversions/property_conversions.go#L150):
+
+``` go
+func CreateTypeConversion(
+    sourceEndpoint *TypedConversionEndpoint,
+    destinationEndpoint *TypedConversionEndpoint,
+    conversionContext *PropertyConversionContext) (PropertyConversion, error)
+```
+
+* `sourceEndpoint` captures the name and type of the source value that needs to be transferred;
+* `destinationEndpoint` similarly captures the name and type of the destination that needs to be populated;
+* `conversionContext` contains other information about the conversion being generated.
+
+
+The core algorithm works by scanning through the handlers listed in [`propertyConversionFactories`](https://github.com/Azure/azure-service-operator/blob/main/v2/tools/generator/internal/conversions/property_conversions.go#L51) to find one that can handle the current conversion. Many of those conversions recursively call `CreateTypeConversion` with a simpler conversion, breaking the problem down into simpler steps.
```

Comment:
`propertyConversionFactories` is the link pointing to the right line? Should it be line 57?